pilot/pkg/xds/delta.go

```diff
@@ -555,13 +553,13 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection,
 // requiresResourceNamesModification checks if a generator needs mutable access to w.ResourceNames.
 // This is used when resources are spontaneously pushed during Delta XDS
 func requiresResourceNamesModification(url string) bool {
-	return url == v3.AddressType
+	return url == v3.AddressType || url == v3.WorkloadType
```

Comment:
for reviewers, workload behave like address, they both use same generator now

Reply:
Is there a reason Workload and Address had different generators to begin with?

Reply:
I mean they use same `WorkloadGenerator`